processor/resourcedetectionprocessor/README.md

```diff
@@ -17,6 +17,8 @@ details of which are currently pending confirmation in the OpenTelemetry specifi
     * host.name
     * os.type
 
+    If running the Collector Docker image, mount `/var/run/docker.sock` and set the `use_docker_sock` flag to `true` to retrieve the attributes from the host machine. Docker detection does not work on macOS.
```

Comment:
I kinda feel like it should be its own detector (you called it out as a possibility in the original issue). Was there any reason you decided not to go that route?

Reply:
> Was there any reason you decided not to go that route?

The main reasons were that 

1.  it felt like the Docker/bare system detectors should evolve together and provide the same attributes when technically feasible: we get the same attributes today and if e.g. someone wants to add the `host.arch` attribute when running the Collector binary it makes sense that they would also want it when running on Docker.
2. the system detector should work regardless of how the Collector is ran: from a user's POV it should make no difference if they are running the Collector from a bare binary or a containerized version. 

But as I wrote, I can definitely see it being its own detector, specially if we think they might end up diverging in some way (e.g. we want to add a `docker.version` attribute or some other attribute that only makes sense on Docker). We have to have a `use_docker_sock` flag anyway so it's not going to make the user experience much worse and it is more flexible

Reply:
@jrcamp I thought about this a bit more, and while I think the arguments above make sense, I think it's likely that these detector end up diverging so I separated it, PTAL